src/test/java/unit/Workspace.java

```diff
@@ -0,0 +1,261 @@
+package unit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.text.IsEqualIgnoringCase.equalToIgnoringCase;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import bio.terra.cli.serialization.userfacing.UFStatus;
+import bio.terra.cli.serialization.userfacing.UFWorkspace;
+import com.fasterxml.jackson.core.type.TypeReference;
+import harness.TestCommand;
+import harness.TestUsers;
+import harness.baseclasses.ClearContextUnit;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra workspace` commands. */
+@Tag("unit")
+public class Workspace extends ClearContextUnit {
+  @Test
+  @DisplayName("status, workspace list reflect workspace create")
+  void statusListReflectCreate() throws IOException {
+    // select a test user and login
+    TestUsers testUser = TestUsers.chooseTestUserWithSpendAccess();
+    testUser.login();
+
+    // `terra workspace create --format=json`
+    UFWorkspace createWorkspace =
+        TestCommand.runCommandExpectSuccess(
+            UFWorkspace.class, "workspace", "create", "--format=json");
+
+    // check the created workspace has an id and a google project
+    assertNotNull(createWorkspace.id, "create workspace returned a workspace id");
+    assertNotNull(createWorkspace.googleProjectId, "create workspace created a gcp project");
+    assertThat(
+        "workspace email matches test user",
```

Comment:
optional style nit: I don't think the comments & messages describing *what* the assertions are doing are especially helpful, especially with the message for the assertion. I think `assertNotNull(createWorkspace.id);` on its own is pretty clear that we're checking that there's a workspace id and there's a lot more to read otherwise. Fine to leave when the assertion is fulfilling a non-obvious purpose without a named helper function, but most of these seem self-explanatory. [go/java-practices/comments](go/java-practices/comments)

Reply:
These messages show up in the stack trace when an assertion fails. I can certainly just look at the line number and then go to the code to see what failed, but I find it helpful to see a readable error message in stdout, particularly when I'm looking at a GH action output and the line numbers don't match exactly with the code I've got on my machine.